app/workers/moderator/sink_articles_worker.rb

```diff
@@ -4,14 +4,15 @@ class SinkArticlesWorker
 
     sidekiq_options queue: :medium_priority, retry: 10
 
-    def perform(user_id)
-      user = User.find_by(id: user_id)
-      return unless user
+    def perform(reaction_id)
+      reaction = Reaction.find_by(id: reaction_id)
+      return unless reaction && reaction.reactable_type == "User"
```

Comment:
this "only sink articles when reaction applies to the user" logic is scattered in the controllers, does it make sense to remove those repeated ad-hoc checks and use this early return? It feels like pushing this into the service (and only enqueuing a job if it's valid, but removing the validation in the moderation/reaction controllers and always passing the reaction to the service) is a small DRY-ing up.